### ISSUE TITLE
EXTERNAL_PROPERTY does not work well with @JsonCreator and FAIL_ON_UNKNOWN_PROPERTIES.

### ISSUE THREAD

**Issue by vdichev** [FasterXML/jackson-databind#754]
When annotating an object with `@JsonCreator` if we use `@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXTERNAL_PROPERTY)` on one of its properties, then deserialization does not work correctly (initializes with nulls) when there are properties which are unknown and are an array or object in the JSON. For examlpe, for this Java code:

``` java
class NetworkUpdate {
    @JsonCreator
    public NetworkUpdate(@JsonProperty("updateContent") Content updateContent) {
        this.updateContent = updateContent;
    }

    Content updateContent;

    public Content getUpdateContent() {
        return updateContent;
    }

    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXTERNAL_PROPERTY, property = "updateType")
    @JsonSubTypes({ @Type(value = Content1.class, name = "BASE"), @Type(value = Content2.class, name = "VIRL") })
    public void setUpdateContent(Content updateContent) {
        this.updateContent = updateContent;
    }

    @Override
    public String toString() {
        return "NetworkUpdate [updateContent=" + updateContent + "]";
    }

}

interface Content {}
class Content1 implements Content {
    @JsonCreator public Content1(@JsonProperty("per") Person per) {
        super();
        this.per = per;
    }

    public Person per; 
}

class Content2 implements Content { 
    @JsonCreator public Content2(@JsonProperty("per") Person per) {
        super();
        this.per = per;
    }

    public Person per; 
}

```

given this JSON:

``` json
{
    "likes": {},
    "updateContent": {"person": {}},
    "updateType": "BASE"
}
```

then if you execute

``` java
ObjectMapper mapper = new ObjectMapper();
mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

and deserialize using this mapper, we find that `updateContent` in `NetworkUpdate` is `null`. If we delete the `likes` field in the JSON or include it in the POJO, it works correctly! Note that the same problem is observed if `likes` is an array, for example an empty one triggers the bug as well: `likes: []`. It is not triggered if `likes` is a simple type like `int` or `String`.

We also experienced the following error when we had an array of these objects: `Unexpected token (END_OBJECT), expected END_ARRAY: expected closing END_ARRAY after type information and deserialized value`

When we debugged the problem, we found that because there was no `propertyBasedCreator` in `com.fasterxml.jackson.databind.deser.BeanDeserialize`, execution was passed to `deserializeWithExternalTypeId` instead of `deserializeUsingPropertyBasedWithExternalTypeId` and the former doesn't call `handleUnknownProperty` (even though their implementations are otherwise very similar).

The issue is reproduced in several versions- at least 2.2.2, 2.4.3, 2.4.5.

**Comment by cowtowncoder** [comment]
Definitely sounds like an interesting bug. Thank you very much for diagnosing the problem. It does sound like deserialization process gets state confused somehow.

**Comment by cowtowncoder** [comment]
Will have a look, thinking this might be related to just fixed #3045.

**Comment by cowtowncoder** [comment]
Cannot reproduce as presented here (non-null value found): assuming fixed by same change that fixed #3045, which will be included in 2.12.2 release.
